core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -146,9 +145,10 @@ public boolean markSupported() {
 
   @Override
   public void close() {
-    // Invalidate the state of the stream.
-    in = null;
-    pos = end;
-    mark = -1;
+    // synchronize on the FSDataInputStream to ensure we are blocked if in the read method
```

Comment:
Maybe an `if (closed) return;` here since we have the volatile `closed` here already (and we can avoid grabbing the lock again).